ember.md

```diff
@@ -0,0 +1,461 @@
+   `tagName` and `classNames` on components and `queryParams` on controllers and
+   routes. Followed by any other simple properties, like default values for properties.
+
+2. __Single line computed property macros__
+
+   E.g. `alias`, `sort` and other macros. Start with service injections. If the
+   module is a model, then `attr` properties should be first, followed by
+   `belongsTo` and `hasMany`.
+
+3. __Multi line computed property functions__
+
+4. __Lifecycle hooks__
+
+   The hooks should be chronologically ordered by the order they are invoked in.
+
+5. __Functions__
+
+   Public functions first, internal functions after.
+
+6. __Actions__
+
+```js
+export default Component.extend({
+  // Plain properties
+  tagName: 'span',
+
+  // Single line CP
+  post: alias('myPost'),
+
+  // Multiline CP
+  authorName: computed('author.{firstName,lastName}', function _authorName() {
+    // code
+  }),
+
+  // Lifecycle hooks
+  didReceiveAttrs() {
+    this._super(...arguments);
+    // code
+  },
+
+  // Functions
+  someFunction() {
+    // code
+  },
+
+  actions: {
+    someAction() {
+      // Code
+    },
+  },
+});
+```
+
+### Override init
+
+Rather than using the object's `init` hook via `on`, override init and
+call `_super` with `...arguments`. This allows you to control execution
+order. [Don't Don't Override Init](https://dockyard.com/blog/2015/10/19/2015-dont-dont-override-init)
+
+## Models
+
+### Organization
+
+Models should be grouped as follows:
+
+* Attributes
+* Associations
+* Computed Properties
+
+Within each section, the attributes should be ordered alphabetically.
+
+```js
+// Good
+
+import Ember from 'ember';
+import DS from 'ember-data';
+
+const { computed } = Ember;
+
+const {
+  attr,
+  hasMany,
+} = DS;
+
+export default DS.Model.extend({
+  // Attributes
+  firstName: attr('string'),
+  lastName: attr('string'),
+
+  // Associations
+  children: hasMany('child'),
+
+  // Computed Properties
+  fullName: computed('firstName', 'lastName', function _fullName() {
+    // Code
+  }),
+});
+
+// Bad
+
+import Ember from 'ember';
+import DS from 'ember-data';
+
+const { computed } = Ember;
+
+const {
+  attr,
+  hasMany,
+} = DS;
+
+export default DS.Model.extend({
+  children: hasMany('child'),
+  firstName: attr('string'),
+  lastName: attr('string'),
+
+  fullName: computed('firstName', 'lastName', function _fullName() {
+    // Code
+  }),
+});
+
+```
+
+## Controllers
+
+### Define query params first
+
+For consistency and ease of discover, list your query params first in
+your controller. These should be listed above default values.
+
+### Alias your model
+
+It provides a cleaner code to name your model `user` if it is a user. It
+is more maintainable, and will fall in line with future routable
+components
+
+```javascript
+export default Ember.Controller.extend({
+  user: computed.readOnly('model'),
+});
+```
+
+## Templates
+
+### Avoid using partials
```

Comment:
Yeah. I already changed this language from the original style guide which said not to use partials at all. I changed it to avoid because we have so many already. However, I think it would be good for us to use components unless a partial is _really_ needed.

This one, like some others, makes it so that a lot of existing code is in violation of the style guide. I think if a best-practice is worth following, which I think this one is, then putting in the style guide will help us as a team know how things _should_ be, even if they aren't that way already. We don't need to drop everything that we are doing to fix all the existing code, but we can write all new code to follow the guidelines and refactor places as we work on them.